utils/utils.py

```diff
@@ -17,3 +19,80 @@ def preprocess_data(X, y=None, remove_zerovar=True, standardize=True):
             X = StandardScaler().fit_transform(X)
 
     return X, y
+
+
+def prox_l1sorted(v, w):
+    """Proximal operator of the OWL norm dot(w, reversed(sort(v)))
+    Follows description and notation from:
+    X. Zeng, M. Figueiredo,
+    The ordered weighted L1 norm: Atomic formulation, dual norm,
+    and projections.
+    eprint http://arxiv.org/abs/1409.4271
+    (From pyowl)
+    XXX
+    """
+
+    v_abs = np.abs(v)
+    ix = np.argsort(v_abs)[::-1]
+    v_abs = v_abs[ix]
+    # project to K+ (monotone non-negative decreasing cone)
+    v_abs = isotonic_regression(v_abs - w, y_min=0, increasing=False)
+
+    # undo the sorting
+    inv_ix = np.zeros_like(ix)
+    inv_ix[ix] = np.arange(len(v))
+    v_abs = v_abs[inv_ix]
+
+    return np.sign(v) * v_abs
+
+
+def prox_slope(beta, lambdas):
```

Comment:
Unless we use JIT compiling via numa using the `@numba` decoration, this version of the prox will be much slower than the other one.